hbase-server/src/main/java/org/apache/hadoop/hbase/io/MetricsIO.java

```diff
@@ -24,10 +24,12 @@
 @InterfaceAudience.Private
 public class MetricsIO {
 
+  private static volatile MetricsIO instance;
   private final MetricsIOSource source;
   private final MetricsIOWrapper wrapper;
 
-  public MetricsIO(MetricsIOWrapper wrapper) {
+  // visible for testing only
```

Comment:
could use the annotation `@VisibleForTesting`"

Reply:
I initially thought to add same but checked code and found this is banned now. See https://github.com/apache/hbase/blob/f6c5dbe97832d4c7d301bce48792365dd2b79e8a/pom.xml#L2535

Reply:
Could use RestrictedApi to restrict the place where we can call this method. It will cause a compilation error if called from other places in the error prone check.

Reply:
Done with commit https://github.com/apache/hbase/pull/5323/commits/0a7b56190e14a1d2ac9cfe044030b2a2dbe0dc72 and updated PR.